src/AppInstallerRepositoryCore/Public/AppInstallerRepositorySearch.h

```diff
@@ -111,6 +111,9 @@ namespace AppInstaller::Repository
         // The highest order field on which the application matched the search.
         ApplicationMatchFilter MatchCriteria;
 
+        // The name of the source where the result is from. Used in aggregated source scenario.
+        std::string SourceName = {};

```

Comment:
>std::string SourceName = {}; [](start = 8, length = 28)

A lot of string duplication with this pattern.  The only solution I see to that is to have this or the IApplication just store a shared_ptr to the source itself.  Not necessarily in need of a change, but maybe some thought.

Reply:
IApplication has a weak pointer to ISource already, then I thought the SearchResult should better be used without having to hold the ISource unless it needs to fetch manifest later.
I also considered making it string_view and store the list of source names somewhere, but it complicates the interface so this ends up as what I can think of here.


---
In reply to: [467338451](https://github.com/microsoft/winget-cli/pull/527#discussion_r467338451) [](ancestors = 467338451)